src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -314,8 +317,13 @@ private void OpenCircuitFor_NeedsLock(Outcome<T> outcome, TimeSpan breakDuration
         scheduledTask = null;
         var utcNow = _timeProvider.GetUtcNow();
 
-        _blockedUntil = IsDateTimeOverflow(utcNow, breakDuration) ? DateTimeOffset.MaxValue : utcNow + breakDuration;
+        if (_breakDurationGenerator is not null && _behavior.FailureCount > 0)
```

Comment:
```suggestion
        if (_breakDurationGenerator is not null)
```

i don't think the second check is needed